langchain/llms/chatglm.py

```diff
@@ -0,0 +1,123 @@
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+from typing import List, Optional, Mapping, Any
+import requests
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+
+class ChatGLM(LLM):
+    """Wrapper around ChatGLM's LLM inference service.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import ChatGLM
+            endpoint_url = (
+                "http://127.0.0.1:8000"
+            )
+            ChatGLM_llm = ChatGLM(
+                endpoint_url=endpoint_url
+            )
+    """
+    
+    endpoint_url: str = (
+        "http://127.0.0.1:8000/"
+    )
+    """Endpoint URL to use."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+    max_token: int = 20000
```

Comment:
Is it possible to support `top_p` in the parameters to get more flexibility?

Reply:
thanks for pointing it out, added.